src/environmentd/src/http.rs

```diff
@@ -308,11 +307,13 @@ async fn auth<B>(
         }
     };
 
+    // Create http default user for local usage.
+    let create_default_user = frontegg.is_none() && tls_mode.is_none() && user == HTTP_DEFAULT_USER;
     // Add the authenticated user as an extension so downstream handlers can
     // inspect it if necessary.
     req.extensions_mut().insert(AuthedUser {
         user,
-        create_if_not_exists: frontegg.is_some(),
+        create_if_not_exists: frontegg.is_some() || create_default_user,
```

Comment:
@benesch Is this what you had in mind for creating the http user on demand?

Reply:
Is it expected ever that `frontegg.is_none()` and the tls mode is `TlsMode::Require`? If so this won't work because `TlsMode::Require` and `frontedd.is_none()` defaults the user to `HTTP_DEFAULT_USER`, but then we won't create the default user because `tls_mode.is_some()`.

Reply:
I'm asking because there are tests in `auth.rs` that explicitly test the HTTP API with `TlsMode::Require` and no frontegg and expect it to work.

Reply:
After discussing with @benesch in person, this condition should be something like `frontegg.is_some() || tls_mode != TlsMode::AssumeUser`.